Lib/unittest/case.py

```diff
@@ -1202,15 +1203,49 @@ def assertIsNot(self, expr1, expr2, msg=None):
             standardMsg = 'unexpectedly identical: %s' % (safe_repr(expr1),)
             self.fail(self._formatMessage(msg, standardMsg))
 
-    def assertDictEqual(self, d1, d2, msg=None):
-        self.assertIsInstance(d1, dict, 'First argument is not a dictionary')
-        self.assertIsInstance(d2, dict, 'Second argument is not a dictionary')
+    def assertDictEqual(self, d1: dict, d2: dict, msg: str | None = None):
```

Comment:
We generally avoid type hints in the stdlib. They are maintained separately over at https://github.com/python/typeshed

Reply:
Removed them.